debezium-core/src/main/java/io/debezium/config/Configuration.java

```diff
@@ -1763,13 +1763,16 @@ default Configuration withReplacedVariables(Function<String, String> valuesByVar
     }
 
     /**
-     * Return a new {@link Configuration} that contains all of the same fields as this configuration, except with masked values
-     * for all keys that end in "password".
+     * Returns a new {@link Configuration} with values masked for all keys matching
+     * the configured sanitize pattern.
+     *
+     * <p>The pattern is taken from {@code custom.sanitize.pattern}, which defaults
+     * to {@link Configuration#PASSWORD_PATTERN} if not set.</p>
      *
      * @return the Configuration with masked values for matching keys; never null
      */
     default Configuration withMaskedPasswords() {
-        return withMasked(PASSWORD_PATTERN);
+        return withMasked(getString("custom.sanitize.pattern"));
```

Comment:
Could you just create a constant for `custom.sanitize.pattern` and also reuse it in the `CUSTOM_SANITIZE_PATTERN` Field?